crypto/s2n_stream_cipher_rc4.c

```diff
@@ -21,20 +21,62 @@
 #include "utils/s2n_blob.h"
 #include "utils/s2n_safety.h"
 
-static uint8_t s2n_stream_cipher_rc4_available()
+#if S2N_OPENSSL_VERSION_AT_LEAST(3, 0, 0)
+    #include "openssl/provider.h"
+DEFINE_POINTER_CLEANUP_FUNC(OSSL_LIB_CTX *, OSSL_LIB_CTX_free);
+#endif
+
+static EVP_CIPHER *s2n_rc4_cipher = NULL;
+
+S2N_RESULT s2n_rc4_init()
+{
+    /* In Openssl-3.0, RC4 is only available from the "legacy" provider,
+     * which is not loaded in the default library context.
+     */
+#if defined(S2N_LIBCRYPTO_SUPPORTS_EVP_RC4) && S2N_OPENSSL_VERSION_AT_LEAST(3, 0, 0)
+    DEFER_CLEANUP(OSSL_LIB_CTX *lib_ctx = OSSL_LIB_CTX_new(), OSSL_LIB_CTX_free_pointer);
+    RESULT_ENSURE_REF(lib_ctx);
+    RESULT_ENSURE_REF(OSSL_PROVIDER_load(lib_ctx, "legacy"));
```

Comment:
Actually realized that the call to `OSSL_PROVIDER_load(NULL, "default")` got dropped. Is that not needed or is it not needed when we load into custom OSSL_LIB_CTX?

Reply:
It's not needed. The default context includes "default" unless you add other providers.

And if it didn't, literally everything would break :)

Reply:
Or do you mean is it needed for RC4? It's not. RC4 is in "legacy", not in "default".